moveit_ros/moveit_servo/src/utils/command.cpp

```diff
@@ -272,6 +273,24 @@ JointDeltaResult jointDeltaFromIK(const Eigen::VectorXd& cartesian_position_delt
     delta_theta = pseudo_inverse * cartesian_position_delta;
   }
 
+  if (!servo_params.active_subgroup.empty())
+  {
+    // Create full delta vector and add delta theta's at actuated joints
+    const auto& move_group_joint_names =
+        robot_state->getJointModelGroup(servo_params.move_group_name)->getActiveJointModelNames();
+    const auto& subgroup_joint_names =
+        robot_state->getJointModelGroup(servo_params.active_subgroup)->getActiveJointModelNames();
+
+    Eigen::VectorXd move_group_delta_theta = Eigen::VectorXd::Zero(move_group_joint_names.size());
```

Comment:
Instead of creating a new vector here, can we just set the required values in `delta_theta` to zero ?

Reply:
Sadly no, because they have different sizes.
move_group_delta_theta.size() = Total number of joints of the whole move group
delta_theta.size() = Number of joints in the subgroup